parsl/executors/taskvine/executor.py

```diff
@@ -549,14 +467,14 @@ def _patch_providers(self):
             if self.project_password_file:
                 self.provider.transfer_input_files.append(self.project_password_file)
 
-    def _serialize_function(self, fn_path, parsl_fn, parsl_fn_args, parsl_fn_kwargs):
+    def _serialize_function(self, app_type, fn_path, parsl_fn, parsl_fn_args, parsl_fn_kwargs):
         """Takes the function application parsl_fn(*parsl_fn_args, **parsl_fn_kwargs)
         and serializes it to the file fn_path."""
 
         # Either build a dictionary with the source of the function, or pickle
         # the function directly:
-        if self.source:
-            function_info = {"source code": inspect.getsource(parsl_fn),
+        if app_type == 'python':
+            function_info = {"source code": autopep8.fix_code(inspect.getsource(parsl_fn)),
```

Comment:
I think in general, you shouldn't expect source code of a function to exist: sometimes callables passed to executors come from places other than raw source (for example, wrappers/closures in monitoring and file staging)

That's also what a bash_app is: just some wrapped python code...

So I think this is generally the wrong thing to be doing by default.

Reply:
I agree that function source code is not a given. This is a bit concerning as the serverless task model in TaskVine assumes otherwise. 

The default behavior now is to transfer source code only if a function object has source code, otherwise we fall back to `pack_apply_message`.

Reply:
does that mean bash_apps don't work with serverless mode?

Reply:
With the current state of serverless task, yes. Serverless tasks use source code to compute a function hash, which is then used to put each function into its own staging directory.

Reply:
ok, that feels like a more fundamental architectural problem which is going to to cause trouble long term with people actually using this for real stuff...

Reply:
It seems that many parts of serverless tasks assume the existence of source code, not just computing a unique directory.

Reply:
Yes tough luck... I'll see if there's any quick fix to this later.

Reply:
ok, so for this PR I recommend getting rid of the changes in core (choosing which decorator) that were made to support this source code stuff, because from the above it looks to me like they're to support a use-case (serverless) which isn't going to work properly...

then let's have a proper discussion about serverless, maybe involving other people - perhaps by email.

(alternatively, if you prefer, rather than removing stuff from this PR, you can open a few new PRs, each with individual changes from this PR that are less controversial to review - that would also provide a path to get much of this work merged, leaving the awkward stuff to think about later)